#!/usr/bin/env python
PACKAGE = "hardware_calibration"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("minimum_steering_feedback", int_t, 0, "An Integer parameter", 192, 0, 1000)
gen.add("maximum_steering_feedback", int_t, 0, "An Integer parameter", 420, 0, 1000)
gen.add("minimum_steering_radians", double_t, 0, "An Integer parameter", 0.512, -3.141, 3.141)
gen.add("maximum_steering_radians", double_t, 0, "An Integer parameter", -0.498, -3.141, 3.141)
gen.add("minimum_steering_pwm", int_t, 0, "An Integer parameter", 950, 700, 2500)
gen.add("maximum_steering_pwm", int_t, 0, "An Integer parameter", 2150, 700, 2500)
gen.add("minimum_speed_pwm", int_t, 0, "An Integer parameter", -1000, -1000, 1000)
gen.add("maximum_speed_pwm", int_t, 0, "An Integer parameter", 1000, -1000, 1000)
gen.add("ticks_to_m", double_t, 0, "An Integer parameter", 0.0027, 0.0, 1.0)
gen.add("number_of_ticks_filter", int_t, 0, "An Integer parameter", 20, 1, 1000)
gen.add("number_of_steering_msgs_filter", int_t, 0, "An Integer parameter", 10, 1, 1000)

exit(gen.generate(PACKAGE, "hardware_calibration", "HardwareCalibration"))
